pkg/channel/fanout/fanout_event_handler.go

```diff
@@ -302,7 +303,6 @@ func (f *FanoutEventHandler) dispatch(ctx context.Context, subs []Subscription,
 			if dispatchResult.err != nil {
 				f.logger.Error("Fanout had an error", zap.Error(dispatchResult.err))
 				dispatchResultForFanout.err = dispatchResult.err
-				return dispatchResultForFanout
```

Comment:
@pierDipi I _think_ this early return may cause a race condition in the synchronous case, as it may lead to the context being cancelled (request completes) before all of the dead letter sink requests have been made if there is more than one error.

However, this means that we won't aggregate the errors and are only showing the final error. Should we add some kind of error aggregation here?

Reply:
Error aggregation sounds good, we could track it into a separate issue